fe/fe-core/src/main/java/com/starrocks/common/profile/Tracers.java

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.common.profile;
+
+import com.google.common.base.Stopwatch;
+import com.starrocks.common.util.RuntimeProfile;
+import com.starrocks.qe.ConnectContext;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.List;
+import java.util.function.Function;
+
+public class Tracers {
+    public enum Mode {
+        NONE, LOGS, VARS, TIMER, TIMING,
+    }
+
+    public enum Module {
+        NONE, ALL, BASE, OPTIMIZER, SCHEDULER, ANALYZE, MV, EXTERNAL,
+    }
+
+    private static final Tracer EMPTY_TRACER = new Tracer() {
+    };
+
+    private static final ThreadLocal<Tracers> THREAD_LOCAL = ThreadLocal.withInitial(Tracers::new);
+
+    private final Tracer[] allTracer = new Tracer[] {EMPTY_TRACER, EMPTY_TRACER};
+
+    private int moduleMask = 0;
+
+    private int modeMask = 0;
+
+    private boolean isCommandLog = false;
+
+    private Tracer tracer(Module module, Mode mode) {
+        return allTracer[(modeMask >> mode.ordinal()) & (moduleMask >> module.ordinal() & 1)];
```

Comment:
Could you add some comments explain to these bit shift operation

Reply:
enable rule is `modeMask |= 1 << mode.ordinal()`, so check rule is `(modeMask >> mode.ordinal()) & 1`

Reply:
what about this `(moduleMask >> module.ordinal() & 1)` ?

Reply:
used to control enable different modules, like `Optimizer`, `MV`......

Reply:
Ok, I get it.